expression/builtin_string_vec.go

```diff
@@ -2868,3 +2866,91 @@ func (b *builtinCharLengthUTF8Sig) vecEvalInt(input *chunk.Chunk, result *chunk.
 	}
 	return nil
 }
+
+func formatDecimal(xBuf *chunk.Column, xDecimals []types.MyDecimal, dInt64s []int64, result *chunk.Column) error {
```

Comment:
How about removing these two functions `formatDecimal` and `formatReal` and let `builtinFormatSig` call `formatDecimalWithLocale` and `formatRealWithLocale` directly?